### PR TITLE
TST: Added regression test for #11395

### PR DESCRIPTION
**Pull request by pimdh**
Added test for #11395 which fails in 1.14 but is fixed in the current master.

### CODE REVIEW

**Comment by eric-wieser** [comment]
Do you mean 1.14?

**Comment by pimdh** [comment]
I do. Have corrected the description.

**Comment by eric-wieser** [comment]
I think it's not just `edge` with the issue - #11011 made this fix to all of `ramp`, `max`, `min`, `med`, `reflect`, `symmetric`, and `wrap`

File: numpy/lib/tests/test_arraypad.py [hunk 1]
```diff
@@ -949,6 +949,15 @@ def test_check_width_shape_1_2(self):
         expected = pad(a, ((1, 2), (1, 2), (1, 2)), 'edge')
         assert_array_equal(padded, expected)
 
+    def test_check_object(self):
+        # Regression test for issue gh-11395.
+        a = np.array([None] * (4 * 3))
+        a = np.reshape(a, (4, 3))
```
Comment (eric-wieser) [inline numpy/lib/tests/test_arraypad.py]: Probably clearer to just write `a = np.full((4, 3), None)`

**Comment by pimdh** [comment]
@eric-wieser I've used `np.full` and extended the tests to `reflect`, `symmetric` and `rwap`. I'm not sure how the `ramp`, `max`, `min` and `med` modes would work on objects.

File: numpy/lib/tests/test_arraypad.py [hunk 2]
```diff
@@ -1009,6 +1009,21 @@ def test_unicode_mode(self):
...
+    def test_object_input(self):
+        # Regression test for issue gh-11395.
+        a = np.full((4, 3), None)
+        pad_amt = ((2, 3), (3, 2))
+        b = np.full((9, 8), None)
+        modes = ['edge',
+                 'symmetric',
+                 'reflect',
+                 'wrap',
+                 ]
```
Comment (eric-wieser) [inline numpy/lib/tests/test_arraypad.py]: Deeply weird style, but I can't fault you for copying what was already in this file

**Comment by eric-wieser** [comment]
Fair enough to leave those - I think you'd want a test on `Decimal` objects for the stats methods.

**Pull request merged**